### PR TITLE
[@smallwei/avue] add types for @smallwei/avue

### PR DESCRIPTION
**Pull request by SoulLyoko**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/smallwei__avue/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// Definitions by: Lyoko <https://github.com/SoulLyoko>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference lib="dom" />
+/// <reference types="./element-plus" />
+/// <reference types="./dic" />
+/// <reference types="./form" />
+/// <reference types="./crud" />
+/// <reference types="./tree" />
+/// <reference types="./global" />
```
Comment (sandersn) [inline types/smallwei__avue/index.d.ts:12]: these don't use module name resolution. I think they should all be `<reference path="./element-plus" />` and so on. I think that would let you get rid of OTHER_FILES.txt too.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/smallwei__avue/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/smallwei__avue/tsconfig.json:18]: this is not allowed, unless there is a very specific reason in changing that line on DT-hosted type declaration:
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/smallwei__avue/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+/// <reference lib="dom" />
+/// <reference path="./element-plus.d.ts" />
+/// <reference path="./dic.d.ts" />
+/// <reference path="./form.d.ts" />
+/// <reference path="./crud.d.ts" />
+/// <reference path="./tree.d.ts" />
+/// <reference path="./global.d.ts" />
+
+// import { App } from 'vue';
+
```
Comment (peterblazejewicz) [inline types/smallwei__avue/index.d.ts:15]:
```suggestion
```
please remove any comments (dead code) that has no meaning in context of the file

File: types/smallwei__avue/element-plus.d.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
+// import type { VNode } from 'vue';
+// import type { RuleItem } from 'async-validator';
```
Comment (peterblazejewicz) [inline types/smallwei__avue/element-plus.d.ts:2]: ditto about dead code imports
you can write proper comments though and keep them inline in the code

File: types/smallwei__avue/form.d.ts [hunk 5]
```diff
@@ -0,0 +1,212 @@
+// import type { Component, VNode } from 'vue';
+
```
Comment (peterblazejewicz) [inline types/smallwei__avue/form.d.ts:2]:
```suggestion
```
ditto

File: types/smallwei__avue/global.d.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+        AvueCrud: typeof AvueCrud;
+        AvueForm: typeof AvueForm;
+        AvueTree: typeof AvueTree;
+    }
+    interface ComponentCustomProperties {
+        $ImagePreview: (list: Array<{ url?: string; thumbUrl?: string }>, index?: number) => void;
+    }
+}
+
+export {};
```
Comment (peterblazejewicz) [inline types/smallwei__avue/global.d.ts:12]: There is no 'global.js` file with a symbols, correct me if I'm wrong:
https://gitee.com/smallweigit/avue/tree/master/src
those TS symbols can be added to 'index.d.ts' file and re-imported in other files here

File: types/smallwei__avue/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    menuType?: 'button' | 'icon' | 'text' | 'menu';
+    theme?: 'dark';
+    axios?: any;
+}
+interface AvueInstaller {
+    install(app: any, config?: AvueConfig): void;
+}
+declare const Avue: AvueInstaller;
+
+export default Avue;
```
Comment (peterblazejewicz) [inline types/smallwei__avue/index.d.ts:27]: the module seems to be webpack exported UMD module:
https://gitee.com/smallweigit/avue/blob/master/build/config.js#L66
https://unpkg.com/browse/@smallwei/avue@2.10.0/lib/avue.js
(L:7 and others)


https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library

**Review by sandersn — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by SoulLyoko** [comment]
@peterblazejewicz What does it mean,the tag "Self Merge"?How can I merge this PR?

**Pull request merged**